TCA Counter App/LockView/LockFeature.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  LockFeature.swift
+//  TCA Counter App
+//
+//  Created by Xiangmu Shi on 2022/7/13.
+//  Copyright © 2022 compass.com. All rights reserved.
+//
+
+import ComposableArchitecture
+import Foundation
+
+struct LockState: Equatable {
+    static func == (lhs: LockState, rhs: LockState) -> Bool {
+        return lhs.countersData == rhs.countersData
+    }
+    var lockAlert: AlertState<LockAction>?
+    var countersData: IdentifiedArrayOf<CounterState> = []
+}
+
+
+enum LockAction {
+    case alertTapped
+    case countStateUpdate(id: CounterState.ID, action: CounterAction)
+}
+
+struct LockEnvironment {
+    static func live() -> LockEnvironment {
+        LockEnvironment(verifyLockNumber: { dataArray in
+            dataArray.map{ $0.count } == [9, 5, 7]
+        })
```

Comment:
Nice and compact 👍 As an exercise, could we refactor this validation so instead of occurring in the environment  it happens in the `lockReducer`?